<a href="https://colab.research.google.com/github/umi-na/google_colab/blob/main/topic_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 토픽 #4: 데이터 전처리 - 데이터 수정, 선택

Colab 실습 시작 전, 이 노트북의 **복사본을 자신의 구글 드라이브에 저장**함: Colab 메뉴에서 "File" >> "Save a Copy in Drive". "Save a Copy" 메뉴 선택 시 "Copy of topic#4.ipynb" 복사본이 구글 드라이브의 'Colab Notebook' 폴더에 저장됨. 그리고 본사본이 새 브라우저 창에 열림.

# 수업목표

* 데이터 읽기 복습 (2주차 수업)

* 데이터 읽기 고급

* 순차적으로 데이터 읽기

* 데이터 필터링 (특정 조건을 만족하는 데이터 읽기; e.g., 나이가 19살 이상인 사람)

* 데이터 수정

# 수업 데이터 소개


## 타이타닉 탑습객 데이터

* 본 수업에서도 [타이타닉 탑습객 데이터](https://drive.google.com/drive/folders/1KpfVquLkXBRByHmpwCvxHYpbpj7sebki?usp=sharing)를 활용하겠음.

* 탑습객 데이터를 다운받아 구글 드라이브에 업로드하여 사용하면 됨

* 탑습객 데이터의 자세한 설명은 [Wk#3 코랩](https://colab.research.google.com/drive/1rn0JqKKVeagdFY-DriMhqIiU2drUeEAg?usp=sharing) 참조



## 구글 드라이브 마운트

* 구글 드라이브에 업로드한 데이터를 사용하기 위해 드라이브를 마운트함

* 아래 그림처럼 (1) => (2) 순으로 클릭하면 드라이브가 마운트됨(자세한 설명은 [Wk#1 코랩](https://colab.research.google.com/drive/1-28UPcDO68IGnwZ0SeuG_jnvMQ--oL6E?usp=sharing) 참조)

![CSV Screenshot](https://drive.google.com/uc?export=view&id=1L56f9PIcVxMCLjonPgqJl2PKzasw4TY8) 

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## 구글 드라이브 폴더 경로 설정

* 아래 root_path의 스트링값은 파일이 있는 구글 드라이브 폴더 경로를 나타냄 

* 자신에게 맞게 구글 드라이브 폴더 경로를 변경하여 사용하면 됨

In [ ]:
root_path = '/content/drive/MyDrive/Colab Notebooks/Colab/wk3_data'

* 아래 데이터는 타이타닉 데이터를 DataFrame으로 읽어와 보여주는 코드임

* 구글 드라이브 마운트 및 폴더 경로가 똑바로 설정되다면 오류 없이 실행됨


In [ ]:
import pandas as pd
from os import path

def getTitanicData():
  return pd.read_csv(path.join(root_path,'titanic.csv'))

df = getTitanicData()
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

# 데이터 읽기 복습 (토픽#2 수업)

* DataFrame의 (1) Column 데이터 (2) Raw 데이터 (3) 단일 데이터를 읽는 방법에 대해 배웠음

* 데이터 필터링를 배위기 전에 배운 내용을 복습하겠음

![CSV Screenshot](https://drive.google.com/uc?export=view&id=1sMOwUG8VgkWLLmgSOhFZTMmXCUfn10nZ) 

## Column 데이터 읽기

* DataFrame[$column\_name$]

* DataFrame[[$column\_name_0$, $column\_name_1$]]


In [ ]:
df['PassengerId']

0      3
1      1
2      3
3      1
4      3
      ..
886    2
887    1
888    3
889    1
890    3
Name: Pclass, Length: 891, dtype: int64

In [ ]:
df[['PassengerId', 'Pclass']]

PassengerId  Pclass
0              1       3
1              2       1
2              3       3
3              4       1
4              5       3
..           ...     ...
886          887       2
887          888       1
888          889       3
889          890       1
890          891       3

[891 rows x 2 columns]


## Row 데이터 읽기

* Row 데이터 읽는 방법은 추후 데이터 필터링에서 사용됨

  * DataFrame[ $index_{시작번호}$ : $index _{끝번호}$ ]

  * DataFrame.loc[ 𝑖𝑛𝑑𝑒𝑥_값} ]

  * DataFrame.loc[ [$index\_값_0$, $index\_값_1$] ]

  * DataFrame.iloc[ 𝑖𝑛𝑑𝑒𝑥_번호 ]
  
  * DataFrame.iloc[ [$index\_번호_0$, $index\_번호_1$]]







* DataFrame[ $index_{시작번호}$ : $index _{끝번호}$ ]

In [ ]:
df[0:4]

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S

[4 rows x 9 columns]

* DataFrame.loc[ 𝑖𝑛𝑑𝑒𝑥_값 ]

* DataFrame.loc[ [$index\_값_0$, $index\_값_1$] ]

In [ ]:
df.loc[0] #location

PassengerId           2
Survived              1
Pclass                1
Sex              female
Age                  38
Ticket         PC 17599
Fare            71.2833
Cabin               C85
Embarked              C
Name: 1, dtype: object

In [ ]:
df.loc[[0,3]]

PassengerId  Survived  Pclass     Sex  ...     Ticket     Fare  Cabin Embarked
0            1         0       3    male  ...  A/5 21171   7.2500    NaN        S
3            4         1       1  female  ...     113803  53.1000   C123        S
5            6         0       3    male  ...     330877   8.4583    NaN        Q

[3 rows x 9 columns]

* DataFrame.iloc[ 𝑖𝑛𝑑𝑒𝑥_번호 ]
  
* DataFrame.iloc[ [$index\_번호_0$, $index\_번호_1$]]


In [ ]:
df.iloc[0]

PassengerId            1
Survived               0
Pclass                 3
Sex                 male
Age                   22
Ticket         A/5 21171
Fare                7.25
Cabin                NaN
Embarked               S
Name: 0, dtype: object

In [ ]:
df.iloc[[0, 3]]

PassengerId  Survived  Pclass     Sex   Age     Ticket   Fare Cabin Embarked
0            1         0       3    male  22.0  A/5 21171   7.25   NaN        S
3            4         1       1  female  35.0     113803  53.10  C123        S

## 단일 데이터 읽기

* DataFrame.at[ $index값$, $column\_name$ ]

* DataFrame.iat[ $index\_번호$, $column\_번호$ ]

In [ ]:
df.at[0,'Ticket']

'A/5 21171'

In [ ]:
df.iat[0, 5]

'A/5 21171'

# 데이터 읽기 고급: 특정 Row 특정 Column 데이터 읽기

* DataFrame.loc 또는 DataFrame.iloc를 사용하여 Row 데이터를 읽는 방법에 대해 배웠음

* DataFrame.loc 또는 DataFrame.iloc은 아래와 같이 개별 row을 보여줄 때 전체 컬럼 값들을 보여줌

PassengerId| Survived| Pclass| Sex| Age| Ticket| Fare| Cabin| Embarked
---| ---| ---| ---| ---| ---| ---| ---| ---
0| 1| 0| 3| male| 22.0| A/5 21171| 7.25| NaN| S

* 전체 컬럼 값들이 아니 특정 컬럼 값들만 읽는 방법을 배우겠음


## DataFrame.loc[인덱스_값, 컬럼_이름]

* DataFrame.loc로 Row의 값을 읽을 때 인덱스_값만 넣어줬음
* 인덱스_값 뒤에 쉼표와 컬럼_이름을 넣어주면 Row데이터를 보여줄 때 해당 컬럼의 데이터 값만 보여줌 


In [ ]:
df = getTitanicData()
df.loc[0, 'Pclass']

3

* 보여주고 싶은 컬럼이 두개 이상이면 리스트로 넣으면 됨

In [ ]:
df.loc[0, ['Pclass', 'Age']]

Pclass     3
Age       22
Name: 0, dtype: object

* 2개 이상의 특정 row의 데이터 값을 보고 싶을 때는 '시작_row : 끝_row'를 넣어주면 됨
* 이 방식은 index가 숫자일 때만 가능함


In [ ]:
df.loc[0:2, ['Pclass', 'Age']]

Pclass   Age
0       3  22.0
1       1  38.0
2       3  26.0

* 특정 row가 아닌 전체 row의 데이터 값을 보고 싶을 때 ':' 를 넣어주면 됨

In [ ]:
df.loc[:, ['Pclass', 'Age']]

Pclass   Age
0         3  22.0
1         1  38.0
2         3  26.0
3         1  35.0
4         3  35.0
..      ...   ...
886       2  27.0
887       1  19.0
888       3   NaN
889       1  26.0
890       3  32.0

[891 rows x 2 columns]

In [ ]:
df.loc[[0, 2], ['Pclass', 'Age']]

Pclass   Age
0       3  22.0
2       3  26.0

## DataFrame.iloc[인덱스_번호, 컬럼_번호]]

* DataFrame.iloc도 DataFrame.loc와 유사하게 사용 가능함

In [ ]:
df = getTitanicData()
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Sex', 'Age', 'Ticket', 'Fare',
       'Cabin', 'Embarked'],
      dtype='object')

In [ ]:
df.iloc[0, 2]

3

In [ ]:
df.iloc[0, [2, 3]]

Pclass       3
Sex       male
Name: 0, dtype: object

In [ ]:
df.iloc[0:2, [2, 3]]

Pclass     Sex
0       3    male
1       1  female

In [ ]:
df.iloc[:, [2, 3]]

Pclass     Sex
0         3    male
1         1  female
2         3  female
3         1  female
4         3    male
..      ...     ...
886       2    male
887       1  female
888       3  female
889       1    male
890       3    male

[891 rows x 2 columns]

In [ ]:
df.iloc[[0, 1], [2, 3]]

Pclass     Sex
0       3    male
1       1  female

# 순차적으로 전체 데이터 읽기

## DataFrame.items

In [ ]:
df = getTitanicData()[:3]
df

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S

[3 rows x 9 columns]

In [ ]:
for label, content in df.items():
  print(label)

PassengerId
Survived
Pclass
Sex
Age
Ticket
Fare
Cabin
Embarked


In [ ]:
df = getTitanicData()[:3]
for label, content in df.iteritems():
  print(content)

0    1
1    2
2    3
Name: PassengerId, dtype: int64
0    0
1    1
2    1
Name: Survived, dtype: int64
0    3
1    1
2    3
Name: Pclass, dtype: int64
0      male
1    female
2    female
Name: Sex, dtype: object
0    22.0
1    38.0
2    26.0
Name: Age, dtype: float64
0           A/5 21171
1            PC 17599
2    STON/O2. 3101282
Name: Ticket, dtype: object
0     7.2500
1    71.2833
2     7.9250
Name: Fare, dtype: float64
0    NaN
1    C85
2    NaN
Name: Cabin, dtype: object
0    S
1    C
2    S
Name: Embarked, dtype: object


## DataFrame.iterrows

In [ ]:
df = getTitanicData()[:3]
for row in df.iterrows():
  print(row)

(0, PassengerId            1
Survived               0
Pclass                 3
Sex                 male
Age                   22
Ticket         A/5 21171
Fare                7.25
Cabin                NaN
Embarked               S
Name: 0, dtype: object)
(1, PassengerId           2
Survived              1
Pclass                1
Sex              female
Age                  38
Ticket         PC 17599
Fare            71.2833
Cabin               C85
Embarked              C
Name: 1, dtype: object)
(2, PassengerId                   3
Survived                      1
Pclass                        3
Sex                      female
Age                          26
Ticket         STON/O2. 3101282
Fare                      7.925
Cabin                       NaN
Embarked                      S
Name: 2, dtype: object)


In [ ]:
df = getTitanicData()[:3]
for index, data in df.iterrows():
  print(index)

0
1
2


In [ ]:
df = getTitanicData()[:3]
for index, data in df.iterrows():
  print(data)

PassengerId            1
Survived               0
Pclass                 3
Sex                 male
Age                   22
Ticket         A/5 21171
Fare                7.25
Cabin                NaN
Embarked               S
Name: 0, dtype: object
PassengerId           2
Survived              1
Pclass                1
Sex              female
Age                  38
Ticket         PC 17599
Fare            71.2833
Cabin               C85
Embarked              C
Name: 1, dtype: object
PassengerId                   3
Survived                      1
Pclass                        3
Sex                      female
Age                          26
Ticket         STON/O2. 3101282
Fare                      7.925
Cabin                       NaN
Embarked                      S
Name: 2, dtype: object


## DataFrame.itertuples

In [ ]:
df = getTitanicData()[:3]
for row in df.itertuples():
  print(row)

Pandas(Index=0, PassengerId=1, Survived=0, Pclass=3, Sex='male', Age=22.0, Ticket='A/5 21171', Fare=7.25, Cabin=nan, Embarked='S')
Pandas(Index=1, PassengerId=2, Survived=1, Pclass=1, Sex='female', Age=38.0, Ticket='PC 17599', Fare=71.2833, Cabin='C85', Embarked='C')
Pandas(Index=2, PassengerId=3, Survived=1, Pclass=3, Sex='female', Age=26.0, Ticket='STON/O2. 3101282', Fare=7.925, Cabin=nan, Embarked='S')


# 데이터 필터링 (특정 부분 데이터 읽기)

* 원하는 조건에 해당하는 데이터만 가져오고 싶을때가 많음. 이럴때 필터링 기능이 유용함

* 필터링을 하는 방법은 다양함. 대표적인 필터링 방법에 대해 배우겠음



## DataFrame[ 조건문 ]

* DataFrame[ $index_{시작번호}$ : $index _{끝번호}$ ]로 데이터 Row를 읽는 법을 배웠음

* $index_{시작번호}$ : $index _{끝번호}$ 대신 조건문을 넣어 필터링을 할 수 있음 (e.g., DataFrame[ 조건문 ])

* 조건문은 참(True) 또는 거짓(False)을 반환함. DataFrame[ 조건문 ]는 조건문이 True를 반환하는 Row만 보여줌 


### 조건문(Conditional Statement) 

* 비교 연산자를 활용하여 조건문을 만들수 있음. 사용 가능한 비교 연산자는 아래 표와 같음

* 비교 연산자는 두 피연산자 (e.g., A vs. B) 사이의 상대적인 크기를 판단함 

비교 연산자|	설명
---| ---
A **==** B|  A와 B는 **같다**
A **!=** B| A와 B는 **다르다**
A **<** B| A는 B 보다 **작다**
A **<=** B| A는 B 보다 **작거나 같다**
A **>** B| A는 B보다 **크다**
A **>=** B| A는 B보다 **크거나 같다** 


### 타이타닉 데이터 필터링

* 타이타닉 데이터를 활용하여 데이터 필터링 하는 법에 대해 배우겠음

* 필터링 조건문(e.g., 생존함 사람만 필터링: Survived == 1)을 설정하기 위해서는 column별 데이터 정보 알고 있어야 함

* 타이나닉 column 정보는 아래표와 같음

컬럼 이름|	설명|	컬럼값 설명
---| ---| ---
PassengerId| 탑승객 번호| 
Survived|	생존유무|	0 = 사망, 1 = 생존
pclass|	티켓 글래스|	1 = 1등석, 2 = 2등석, 3 = 3등석
Sex|	성별| Female = 여성, Male = 남성	
Age|	만 나이 | 태어난지 1년 미만으면(만 나이 < 0) 소수로 표현
Ticket|	티켓 번호|	
Fare|	티켓 가격|	
Cabin|	객실 번호|	
Embarked|	탑승 항구|	C = 셰르부르(프랑스 서북부 항구), Q = 퀸즈타운(아일랜드 남부의 항구), S = 사우샘프턴 (영국 남부 항구)


In [ ]:
df = getTitanicData()
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]


* Survived 컬럼은 탑승객의 생존유무를 나타남. Survived 정보를 활용하여 생존한 탑승객만 나타나도록 필터링하겠음

* 생존자는 Survived 컬럼값이 '1'임. 따라서 'df.Survived이 == 1'라는 조건문을 만들면 됨.

* 'df.Survived == 1'는 Survied 컬럼값이 1일때 True을 반환함

In [ ]:
df.Survived == 1

0      False
1       True
2       True
3       True
4      False
       ...  
886    False
887     True
888    False
889     True
890    False
Name: Survived, Length: 891, dtype: bool

* 참고로 df.Survived 는 df['Survived']처럼 Survived 컬럼 데이터를 보여줌

In [ ]:
df.Survived.head()

0    0
1    1
2    1
3    1
4    0
Name: Survived, dtype: int64

In [ ]:
df['Survived'].head()

0    0
1    1
2    1
3    1
4    0
Name: Survived, dtype: int64

* 추후 df[df.Survived == 1] 실행 시 조건문이 True인 Row만 보여주는지 알아보기 위해 조건문 반환값을 보여주는 컬럼을 만들겠음

* 아래 코드는 'isSurvived'이라는 새로운 컬럼을 생성 후 'df.Survived == 1'의 반환값(i.e., True/False)을 데이터로 넣음

* isSurvived 컬럼은 DataFrame의 가장 마지막에 있음

In [ ]:
df['isSurvived'] = df.Survived == 1
df

PassengerId  Survived  Pclass     Sex  ...     Fare Cabin  Embarked isSurvived
0              1         0       3    male  ...   7.2500   NaN         S      False
1              2         1       1  female  ...  71.2833   C85         C       True
2              3         1       3  female  ...   7.9250   NaN         S       True
3              4         1       1  female  ...  53.1000  C123         S       True
4              5         0       3    male  ...   8.0500   NaN         S      False
..           ...       ...     ...     ...  ...      ...   ...       ...        ...
886          887         0       2    male  ...  13.0000   NaN         S      False
887          888         1       1  female  ...  30.0000   B42         S       True
888          889         0       3  female  ...  23.4500   NaN         S      False
889          890         1       1    male  ...  30.0000  C148         C       True
890          891         0       3    male  ...   7.7500   NaN         Q      False

[891 rows x 10 columns]

* 아래 코드의 결과를 보면 df['df.Survived == 1']는 'df.Survived == 1'가 True를 가지는 Row만 보여줌을 알수 있음

* 윗 코드의 결과를 보면 isSurvived 컬럼에 True와 False 모두가 나와 있지만 아래 결과를 보면 True만 있음

In [ ]:
df[df.Survived == 1]

PassengerId  Survived  Pclass     Sex  ...     Fare Cabin  Embarked isSurvived
1              2         1       1  female  ...  71.2833   C85         C       True
2              3         1       3  female  ...   7.9250   NaN         S       True
3              4         1       1  female  ...  53.1000  C123         S       True
8              9         1       3  female  ...  11.1333   NaN         S       True
9             10         1       2  female  ...  30.0708   NaN         C       True
..           ...       ...     ...     ...  ...      ...   ...       ...        ...
875          876         1       3  female  ...   7.2250   NaN         C       True
879          880         1       1  female  ...  83.1583   C50         C       True
880          881         1       2  female  ...  26.0000   NaN         S       True
887          888         1       1  female  ...  30.0000   B42         S       True
889          890         1       1    male  ...  30.0000  C148         C       True

[342 rows x 10 columns]

## DataFrame[ 조건문$_A$ & 조건문$_B$ ]

* 조건문을 한개 이상 넣어 필터링을 할 수 있음 (e.g., DataFrame[ 조건문 & 조건문 ])

* 조건문이 두개 이상일 때는 조건문 사이에 논리 연산자를 넣어야 함. 그리고 개별 조건문은 괄호안에 들어었어야 함: e.g., df[(df.Sex == 'male') & (20 <= df.Age )]


#### 논리 연산자(Logical Operator)

* 논리 연산자를 두 조건문 사이에 넣어 조건문을 더 다양하게 만들 수 있음: e.g., (A == B) & (B != C)

* 논리 연산자로 다음 3가지를 사용할 수 있음: & (AND; 그리고), | (OR; 또는), ! (NOT; 아니다)

* 논리 연산자 사용예는 아래표와 같음


논리 연산자| 설명| 사용 예시| 설명
---| --- | --- | ---
& | (A == B) & (B != C) | 앞뒤 모든 조건문을 만족해야 참이됨 |A와 B는 같고 B는 C와 다르다
\| | (A == B) \| (B != C) | 앞뒤 조건문 중 한개만 만족하면 참이됨|A와 B는 같거나 B는 C와 다르다
! | !(A == B) | 조건문 거짓이면 참이됨 |A와 B는 같은 것이 아니다 (A와 B는 다르다)

#### 타이타닉 데이터 필터링


In [ ]:
df = getTitanicData()
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

* 타이타닉 데이터에서 3등석 탑승객 중 생존한 탑승객만 찾고 싶음

* 즉 티켓 글래스가 3등석(df.Pclass == 3)이고(&) 생존유무는 생존인 탑승객(df.Survived == 1)임

In [ ]:
df[(df.Pclass == 3) & (df.Survived == 1)]

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
2              3         1       3  ...   7.9250    NaN        S
8              9         1       3  ...  11.1333    NaN        S
10            11         1       3  ...  16.7000     G6        S
19            20         1       3  ...   7.2250    NaN        C
22            23         1       3  ...   8.0292    NaN        Q
..           ...       ...     ...  ...      ...    ...      ...
838          839         1       3  ...  56.4958    NaN        S
855          856         1       3  ...   9.3500    NaN        S
858          859         1       3  ...  19.2583    NaN        C
869          870         1       3  ...  11.1333    NaN        S
875          876         1       3  ...   7.2250    NaN        C

[119 rows x 9 columns]

* 다음으로 타이타틱 생존자이면서 여성이고 나이가 40살 이상이였던 탑승객을 찾아보겠음

In [ ]:
result = df[(df.Survived == 1) & (df.Sex == 'female') & (40 <= df.Age)]
result

PassengerId  Survived  Pclass  ...      Fare    Cabin Embarked
11            12         1       1  ...   26.5500     C103        S
15            16         1       2  ...   16.0000      NaN        S
52            53         1       1  ...   76.7292      D33        C
161          162         1       2  ...   15.7500      NaN        S
194          195         1       1  ...   27.7208       B4        C
195          196         1       1  ...  146.5208      B80        C
259          260         1       2  ...   26.0000      NaN        S
268          269         1       1  ...  153.4625     C125        S
272          273         1       2  ...   19.5000      NaN        S
275          276         1       1  ...   77.9583       D7        S
299          300         1       1  ...  247.5208  B58 B60        C
319          320         1       1  ...  134.5000      E34        C
337          338         1       1  ...  134.5000      E40        C
346          347         1       2  ...   13.0000      NaN        S
366          367         1       1  ...   75.2500      D37        C
380          381         1       1  ...  227.5250      NaN        C
432          433         1       2  ...   26.0000      NaN        S
440          441         1       2  ...   26.2500      NaN        S
458          459         1       2  ...   10.5000      NaN        S
483          484         1       3  ...    9.5875      NaN        S
496          497         1       1  ...   78.2667      D20        C
513          514         1       1  ...   59.4000      NaN        C
523          524         1       1  ...   57.9792      B18        C
526          527         1       2  ...   10.5000      NaN        S
556          557         1       1  ...   39.6000      A16        C
571          572         1       1  ...   51.4792     C101        S
591          592         1       1  ...   78.2667      D20        C
609          610         1       1  ...  153.4625     C125        S
670          671         1       2  ...   39.0000      NaN        S
706          707         1       2  ...   13.5000      NaN        S
754          755         1       2  ...   65.0000      NaN        S
765          766         1       1  ...   77.9583      D11        S
774          775         1       2  ...   23.0000      NaN        S
779          780         1       1  ...  211.3375       B3        S
796          797         1       1  ...   25.9292      D17        S
820          821         1       1  ...   93.5000      B69        S
829          830         1       1  ...   80.0000      B28      NaN
856          857         1       1  ...  164.8667      NaN        S
862          863         1       1  ...   25.9292      D17        S
865          866         1       2  ...   13.0000      NaN        S
871          872         1       1  ...   52.5542      D35        S
879          880         1       1  ...   83.1583      C50        C

[42 rows x 9 columns]

* 필터링 결과로 총 42명의 탑승객이 나왔음

In [ ]:
len(result)

42

* 다음으로 타이타틱 생존자이면서 여성이고 나이가 40살 이상이였던 탑승객 중 2등석 탑승객을 찾아보겠음

* 바로 전 DataFrame 결과는 " 타이아틱 생존자이면서 여성이고 나이가 40살 이상이였던 탑승객"을 보여줌.이 DataFrame를 재활용해서 필터링 하겠음 

In [ ]:
result[result.Pclass == 2]

PassengerId  Survived  Pclass     Sex  ...        Ticket   Fare  Cabin Embarked
15            16         1       2  female  ...        248706  16.00    NaN        S
161          162         1       2  female  ...    C.A. 33595  15.75    NaN        S
259          260         1       2  female  ...        230433  26.00    NaN        S
272          273         1       2  female  ...        250644  19.50    NaN        S
346          347         1       2  female  ...         31418  13.00    NaN        S
432          433         1       2  female  ...    SC/AH 3085  26.00    NaN        S
440          441         1       2  female  ...  F.C.C. 13529  26.25    NaN        S
458          459         1       2  female  ...  F.C.C. 13531  10.50    NaN        S
526          527         1       2  female  ...   W./C. 14258  10.50    NaN        S
670          671         1       2  female  ...         29750  39.00    NaN        S
706          707         1       2  female  ...        223596  13.50    NaN        S
754          755         1       2  female  ...        220845  65.00    NaN        S
774          775         1       2  female  ...         29105  23.00    NaN        S
865          866         1       2  female  ...        236852  13.00    NaN        S

[14 rows x 9 columns]

* 다음으로 세르부르 또는 쿼즈타운에서 탑승한 승객 중 생존자를 찾겠음

In [ ]:
df[((df.Embarked == 'C') | (df.Embarked == 'Q')) & (df.Survived == 1) ]

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
1              2         1       1  ...  71.2833    C85        C
9             10         1       2  ...  30.0708    NaN        C
19            20         1       3  ...   7.2250    NaN        C
22            23         1       3  ...   8.0292    NaN        Q
28            29         1       3  ...   7.8792    NaN        Q
..           ...       ...     ...  ...      ...    ...      ...
866          867         1       2  ...  13.8583    NaN        C
874          875         1       2  ...  24.0000    NaN        C
875          876         1       3  ...   7.2250    NaN        C
879          880         1       1  ...  83.1583    C50        C
889          890         1       1  ...  30.0000   C148        C

[123 rows x 9 columns]

## DataFrame.loc[조건문]

* DataFrame.loc 메소스을 사용하여 필터링 하는 법을 배우겠음. DataFrame.loc[ 𝑖𝑛𝑑𝑒𝑥_값 ]로 데이터 Row를 읽는 법을 배웠음

* 𝑖𝑛𝑑𝑒𝑥_값 대신 조건문을 넣어 필터링을 할 수 있음 (e.g., DataFrame.loc[ 조건문 ])

* 아래 코드는 생존한 타이타닉 탑습객을 필터하여 보여줌

In [ ]:
df = getTitanicData()
df.loc[df.Survived == 1]

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
1              2         1       1  ...  71.2833    C85        C
2              3         1       3  ...   7.9250    NaN        S
3              4         1       1  ...  53.1000   C123        S
8              9         1       3  ...  11.1333    NaN        S
9             10         1       2  ...  30.0708    NaN        C
..           ...       ...     ...  ...      ...    ...      ...
875          876         1       3  ...   7.2250    NaN        C
879          880         1       1  ...  83.1583    C50        C
880          881         1       2  ...  26.0000    NaN        S
887          888         1       1  ...  30.0000    B42        S
889          890         1       1  ...  30.0000   C148        C

[342 rows x 9 columns]

* 다음 코드는 3등석 탑승객 중 생존한 탑승객만 필터하여 보여줌

In [ ]:
df.loc[(df.Pclass == 3) & (df.Survived == 1)]

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
2              3         1       3  ...   7.9250    NaN        S
8              9         1       3  ...  11.1333    NaN        S
10            11         1       3  ...  16.7000     G6        S
19            20         1       3  ...   7.2250    NaN        C
22            23         1       3  ...   8.0292    NaN        Q
..           ...       ...     ...  ...      ...    ...      ...
838          839         1       3  ...  56.4958    NaN        S
855          856         1       3  ...   9.3500    NaN        S
858          859         1       3  ...  19.2583    NaN        C
869          870         1       3  ...  11.1333    NaN        S
875          876         1       3  ...   7.2250    NaN        C

[119 rows x 9 columns]

* 다음 코드는 3등석 탑승객 중 티켓을 30파운드 이하로 구입한 탑승객을 필터한 후 Survived 컬럼 데이터를 보여줌

In [ ]:
result = df.loc[(df.Pclass == 3) & (df.Fare <= 30)]
result.Survived

0      0
2      1
4      0
5      0
7      0
      ..
882    0
884    0
885    0
888    0
890    0
Name: Survived, Length: 450, dtype: int64

## DataFrame.loc[조건문, 컬럼이름]

* DataFrame[조건문]은 필터 후 모든 컬럼의 데이터 값을 보여줌

* DataFrame[조건문]과 비교하여 DataFrame.loc 메소드 사용 시 가장 큰 이점은 필터 후 보여주고 싶은 column을 선택할 수 있음
* 예컨데 3등석 탑승객 중 생존한 탑승객을 필터 후 Age 컬럼 데이터만 보여줄 수 있음 

In [ ]:
df[(df.Pclass == 3) & (df.Survived == 1)]

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
2              3         1       3  ...   7.9250    NaN        S
8              9         1       3  ...  11.1333    NaN        S
10            11         1       3  ...  16.7000     G6        S
19            20         1       3  ...   7.2250    NaN        C
22            23         1       3  ...   8.0292    NaN        Q
..           ...       ...     ...  ...      ...    ...      ...
838          839         1       3  ...  56.4958    NaN        S
855          856         1       3  ...   9.3500    NaN        S
858          859         1       3  ...  19.2583    NaN        C
869          870         1       3  ...  11.1333    NaN        S
875          876         1       3  ...   7.2250    NaN        C

[119 rows x 9 columns]

In [ ]:
df.loc[(df.Pclass == 3) & (df.Survived == 1), 'Age']

2      26.0
8      27.0
10      4.0
19      NaN
22     15.0
       ... 
838    32.0
855    18.0
858    24.0
869     4.0
875    15.0
Name: Age, Length: 119, dtype: float64

* 필터 후 보여주고 싶은 컬럼이 두개 이상이면 리스트를 넣어주면 됨

In [ ]:
df.loc[(df.Pclass == 3) & (df.Survived == 1), ['Pclass','Survived','Age']]

Pclass  Survived   Age
2         3         1  26.0
8         3         1  27.0
10        3         1   4.0
19        3         1   NaN
22        3         1  15.0
..      ...       ...   ...
838       3         1  32.0
855       3         1  18.0
858       3         1  24.0
869       3         1   4.0
875       3         1  15.0

[119 rows x 3 columns]

# 데이터 수정



## 단일 데이터 수정


### DataFrame.at / DataFrame.iat

* DataFrame.at 또는 DataFrame.iat를 사용하여 단일 데이터를 읽어오는 방법에 대해 배웠음

* DataFrame.at 또는 DataFrame.iat를 사용하여 단일 데이터를 수정할 수 있음

#### DataFrame.at[인덱스_값, 컬럼_이름]

* DataFrame.at[인덱스_값, 컬럼_이름]를 사용하여 index = 2, column = 'Pclass' 데이터를 수정겠음

* 수정 전, 먼저 index = 2, column = 'Pclass' 데이터 값을 확인 하겠음

In [ ]:
df = getTitanicData()
df.at[2, 'Pclass']

3

* index = 2, column = 'Pclass' 데이터 값은 3이였음. 이 값을 4로 수정한 후 해당 데이터 값을 확인 하겠음

In [ ]:
df.at[2, 'Pclass'] = 4
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       4  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

#### DataFrame.iat[인덱스_번호, 컬럼_번호]

* DataFrame.iat[인덱스_번호, 컬럼_번호]를 사용하여 index_no = 2, column_no = 2 데이터를 4로 수정하겠음

In [ ]:
df = getTitanicData()
df.iat[2, 2]

3

In [ ]:
df.iat[2, 2] = 4
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       4  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

### DataFrame.loc / DataFrame.iloc

#### DataFrame.loc[인덱스_값, 컬럼_이름]

* DataFrame.loc[인덱스_값, 컬럼_이름]를 사용하여 index = 0, column = 'Pclass' 데이터를 수정겠음

* 수정 전, 먼저 index = 2, column = 'Pclass' 데이터 값을 확인 하겠음

In [ ]:
df = getTitanicData()
df.loc[0, 'Pclass']

3

In [ ]:
df.loc[0, 'Pclass'] = 4
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       4  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

#### DataFrame.iloc[인덱스_번호, 컬럼_번호]

* DataFrame.loc[인덱스_번호, 컬럼_번호]를 사용하여 index_no = 0, column_no = 2 데이터를 4로 수정하겠음

In [ ]:
df = getTitanicData()
df.iloc[0, 2]

22.0

In [ ]:
df.iloc[0, 2] = 4
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

## Column 수정

### DataFrame[컬럼_이름]

* DataFrame[컬럼_이름]을 사용하여 해당 이름을 가진 컬럼 데이터를 읽는 법을 배웠음
* DataFrame[컬럼_이름]에 원하는 값을 할당하여 컬럼 전체 데이터를 수정됨
* Pclass 컬럼 전체 데이터를 수정하여 보겠음. 수정 전에 Pclass 컬럼 데이터를 먼저 확인하겠음

In [ ]:
df = getTitanicData()
df['Pclass']

0      3
1      1
2      3
3      1
4      3
      ..
886    2
887    1
888    3
889    1
890    3
Name: Pclass, Length: 891, dtype: int64

* Pclass 컬럼 데이터를 4로 수정한 후 Pclass 컬럼 데이터를 확인하겠음

In [ ]:
df['Pclass'] = 4
df['Pclass'] 

0      4
1      4
2      4
3      4
4      4
      ..
886    4
887    4
888    4
889    4
890    4
Name: Pclass, Length: 891, dtype: int64

### DataFrame[ [ 컬럼_이름$_0$ ,  컬럼_이름$_1$ ] ]

* DataFrame[ [ 컬럼_이름$_0$ ,  컬럼_이름$_1$ ] ]은 컬럼 이름을 리스트로 넣으면 해당 컬럼들의  데이터를 읽을 수 있음

* DataFrame[ [ 컬럼_이름$_0$ ,  컬럼_이름$_1$ ] ]에 원하는 값을 할당하면 해당 컬럼들의 데이터가 수정됨

* Pclass와 Sex 컬럼 데이터를 수정하여 보겠음



In [ ]:
df = getTitanicData()
df[['Pclass', 'Sex']]

Pclass     Sex
0         3    male
1         1  female
2         3  female
3         1  female
4         3    male
..      ...     ...
886       2    male
887       1  female
888       3  female
889       1    male
890       3    male

[891 rows x 2 columns]

In [ ]:
df[['Pclass', 'Sex']] = 4
df[['Pclass', 'Sex']]

Pclass  Sex
0         4    4
1         4    4
2         4    4
3         4    4
4         4    4
..      ...  ...
886       4    4
887       4    4
888       4    4
889       4    4
890       4    4

[891 rows x 2 columns]

### DataFrame.loc[ : , 컬럼이름]

* DataFrame.loc[ : , 컬럼이름]을 사용하여 특정 컬럼의 전체 데이터를 읽을 수 있음

* DataFrame.loc[ : , 컬럼이름]에 원하는 값을 할당하면 해당 컬럼의 데이터가 수정됨

* Pclass 컬럼 데이터를 수정하여 보겠음

In [ ]:
df = getTitanicData()
df.loc[:, 'Pclass']

0      3
1      1
2      3
3      1
4      3
      ..
886    2
887    1
888    3
889    1
890    3
Name: Pclass, Length: 891, dtype: int64

In [ ]:
df.loc[:, 'Pclass'] = 4
df.loc[:, 'Pclass']

0      4
1      4
2      4
3      4
4      4
      ..
886    4
887    4
888    4
889    4
890    4
Name: Pclass, Length: 891, dtype: int64

### DataFrame.loc[ : , [컬럼_이름$_0$ ,  컬럼_이름$_1$]]

* DataFrame.loc[ : , [컬럼_이름$_0$ ,  컬럼_이름$_1$]]을 사용하여 특정 컬럼들의 전체 데이터를 읽을 수 있음

* DataFrame.loc[ : , [컬럼_이름$_0$ ,  컬럼_이름$_1$]]에 원하는 값을 할당하면 해당 컬럼의 데이터가 수정됨

* Pclass와 Age 컬럼 데이터를 수정하여 보겠음

In [ ]:
df = getTitanicData()
df.loc[:, ['Pclass', 'Age']]

Pclass   Age
0         3  22.0
1         1  38.0
2         3  26.0
3         1  35.0
4         3  35.0
..      ...   ...
886       2  27.0
887       1  19.0
888       3   NaN
889       1  26.0
890       3  32.0

[891 rows x 2 columns]

In [ ]:
df.loc[:, ['Pclass', 'Age']] = 4
df.loc[:, ['Pclass', 'Age']]

Pclass  Age
0         4    4
1         4    4
2         4    4
3         4    4
4         4    4
..      ...  ...
886       4    4
887       4    4
888       4    4
889       4    4
890       4    4

[891 rows x 2 columns]

## Row 수정

### DataFrame[인덱트_시작번호 : 인텍스_끝번호]

* DataFrame[인덱트_시작번호 : 인텍스_끝번호]로 해당 범위의 Row를 읽을 수 있음
* DataFrame[인덱트_시작번호 : 인텍스_끝번호]에 원하는 값을 할당하면 해당 Row 데이터가 수정됨

In [ ]:
df = getTitanicData()
df[0:2]

PassengerId  Survived  Pclass     Sex  ...     Ticket     Fare  Cabin Embarked
0            1         0       3    male  ...  A/5 21171   7.2500    NaN        S
1            2         1       1  female  ...   PC 17599  71.2833    C85        C

[2 rows x 9 columns]

In [ ]:
df[0:2] = 1
df[0:2]

PassengerId  Survived  Pclass Sex  Age Ticket  Fare Cabin Embarked
0            1         1       1   1    1      1   1.0     1        1
1            1         1       1   1    1      1   1.0     1        1

### DataFrame.loc[인덱스_값]


In [ ]:
df = getTitanicData()
df.loc[0]

PassengerId            1
Survived               0
Pclass                 3
Sex                 male
Age                   22
Ticket         A/5 21171
Fare                7.25
Cabin                NaN
Embarked               S
Name: 0, dtype: object

In [ ]:
df.loc[0] = 3
df.loc[0]

PassengerId    3
Survived       3
Pclass         3
Sex            3
Age            3
Ticket         3
Fare           3
Cabin          3
Embarked       3
Name: 0, dtype: object

In [ ]:
df.head()

PassengerId  Survived  Pclass  ...    Fare  Cabin Embarked
0            3         3       3  ...   3.000      3        3
1            1         1       1  ...   1.000      1        1
2            3         1       4  ...   7.925    NaN        S
3            4         1       4  ...  53.100   C123        S
4            5         0       4  ...   8.050    NaN        S

[5 rows x 9 columns]

In [ ]:
df.head()

PassengerId  Survived  Pclass  ...    Fare  Cabin Embarked
0            NaN       NaN     NaN  ...     NaN    NaN      NaN
1            1.0       1.0     1.0  ...   1.000      1        1
2            3.0       1.0     4.0  ...   7.925    NaN        S
3            4.0       1.0     4.0  ...  53.100   C123        S
4            5.0       0.0     4.0  ...   8.050    NaN        S
..           ...       ...     ...  ...     ...    ...      ...
886        887.0       0.0     4.0  ...  13.000    NaN        S
887        888.0       1.0     4.0  ...  30.000    B42        S
888        889.0       0.0     4.0  ...  23.450    NaN        S
889        890.0       1.0     4.0  ...  30.000   C148        C
890        891.0       0.0     4.0  ...   7.750    NaN        Q

[891 rows x 9 columns]

## Row 범위 + Column 범위 데이터 수정

* DataFrame.loc[인덱스_값_시작:인덱스_값_끝, [컬럼_이름]]을 사용하여 해당 Row 범위와 Column 범위의 데이터만 읽을 수 있음

* DataFrame.loc[인덱스_값_시작:인덱스_값_끝, [컬럼_이름]]에 원하는 값을 할당하면 해당 Row 범위와 Column 범위의 데이터만 값이 수정됨

* 이 방법은 index가 숫자 일때만 사용 가능함. 숫자가 아니라면 DataFrame.iloc을 사용하여야 함

In [ ]:
df = getTitanicData()
df.loc[0:2, 'Pclass']

0    3
1    1
2    3
Name: Pclass, dtype: int64

In [ ]:
df.loc[0:2, 'Pclass'] = 4
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       4  ...   7.2500    NaN        S
1            2         1       4  ...  71.2833    C85        C
2            3         1       4  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

* 컬럼 이름이 두개 이상이면 컬럼 이름을 list로 넣어주면 됨

In [ ]:
df = getTitanicData()
df.loc[0:2, ['Pclass', 'Sex']]

Pclass     Sex
0       3    male
1       1  female
2       3  female

In [ ]:
df.loc[0:2, ['Pclass', 'Sex']] = 3
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       3  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

* 시작:끝 범위 대신 row값을 list로 넣어주면 됨

In [ ]:
df = getTitanicData()
df.loc[[0,2], ['Pclass', 'Sex']]

Pclass     Sex
0       3    male
2       3  female

In [ ]:
df.loc[[0,2], ['Pclass', 'Sex']] = 3
df.head()

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0            1         0       3  ...   7.2500    NaN        S
1            2         1       1  ...  71.2833    C85        C
2            3         1       3  ...   7.9250    NaN        S
3            4         1       1  ...  53.1000   C123        S
4            5         0       3  ...   8.0500    NaN        S

[5 rows x 9 columns]

## 필터링를 활용한 데이터 수정 

* 위에서 필터링 하는 방법을 배웠음. 필터링 결과에 원하는 값을 할당하면 해당 데이터만 수정됨



### DataFrame[조건문]

* DataFrame[조건문]은 조건문이 True를 반환하는 Row만 보여줌

* DataFrame[조건문] 결과에 원하는 데이터를 할당하면 반환되는 Row 데이터가 수정됨

* 생존자 (df.Survived == 1) 데이터를 모두 3으로 수정하여 보겠음. 수정 전에 생존자 데이터를 먼저 확인해 보겠음

In [ ]:
df = getTitanicData()
df[df.Survived == 1]

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
1              2         1       1  ...  71.2833    C85        C
2              3         1       3  ...   7.9250    NaN        S
3              4         1       1  ...  53.1000   C123        S
8              9         1       3  ...  11.1333    NaN        S
9             10         1       2  ...  30.0708    NaN        C
..           ...       ...     ...  ...      ...    ...      ...
875          876         1       3  ...   7.2250    NaN        C
879          880         1       1  ...  83.1583    C50        C
880          881         1       2  ...  26.0000    NaN        S
887          888         1       1  ...  30.0000    B42        S
889          890         1       1  ...  30.0000   C148        C

[342 rows x 9 columns]

* 생존자 데이터를 3으로 수정하였음. 따라서 df.Survived == 1 조건문에 해당하는 데이터는 없음

In [ ]:
df[df.Survived == 1] = 3
df[df.Survived == 1]

Empty DataFrame
Columns: [PassengerId, Survived, Pclass, Sex, Age, Ticket, Fare, Cabin, Embarked]
Index: []

* df.Survived == 1이 였던 데이터들은 모두 3으로 변경되었음. 따라서 df.Survived == 3으로 확인해 보겠음

In [ ]:
df[df.Survived == 3] 

PassengerId  Survived  Pclass Sex  Age Ticket  Fare Cabin Embarked
1              3         3       3   3  3.0      3   3.0     3        3
2              3         3       3   3  3.0      3   3.0     3        3
3              3         3       3   3  3.0      3   3.0     3        3
8              3         3       3   3  3.0      3   3.0     3        3
9              3         3       3   3  3.0      3   3.0     3        3
..           ...       ...     ...  ..  ...    ...   ...   ...      ...
875            3         3       3   3  3.0      3   3.0     3        3
879            3         3       3   3  3.0      3   3.0     3        3
880            3         3       3   3  3.0      3   3.0     3        3
887            3         3       3   3  3.0      3   3.0     3        3
889            3         3       3   3  3.0      3   3.0     3        3

[342 rows x 9 columns]

### DataFrame.loc[조건문, 컬럼이름] 
* 데이터 필터링 후 전체 데이터가 아닌 특정 column만 수정하는 방법에 대해 배우겠음

* DataFrame.loc[조건문, 컬럼이름]으로 데이터 필터링 후 특정 column 데이터만 읽을 수 있음

* DataFrame.loc[조건문, 컬럼이름]에 원하는 값을 할당하면 해당 column 데이터만 수정됨

In [ ]:
df = getTitanicData()
df.loc[df.Survived == 0, 'Pclass']

0      3
4      3
5      3
6      1
7      3
      ..
884    3
885    3
886    2
888    3
890    3
Name: Pclass, Length: 549, dtype: int64

In [ ]:
df.loc[df.Survived == 0, 'Pclass'] = 4
df.loc[df.Survived == 0, 'Pclass']

0      4
4      4
5      4
6      4
7      4
      ..
884    4
885    4
886    4
888    4
890    4
Name: Pclass, Length: 549, dtype: int64

In [ ]:
df

PassengerId  Survived  Pclass  ...     Fare  Cabin Embarked
0              1         0       4  ...   7.2500    NaN        S
1              2         1       1  ...  71.2833    C85        C
2              3         1       3  ...   7.9250    NaN        S
3              4         1       1  ...  53.1000   C123        S
4              5         0       4  ...   8.0500    NaN        S
..           ...       ...     ...  ...      ...    ...      ...
886          887         0       4  ...  13.0000    NaN        S
887          888         1       1  ...  30.0000    B42        S
888          889         0       4  ...  23.4500    NaN        S
889          890         1       1  ...  30.0000   C148        C
890          891         0       4  ...   7.7500    NaN        Q

[891 rows x 9 columns]

* 이번에는 2개의 컬럼 데이터를 수정하여 보겠음

In [ ]:
df = getTitanicData()
df.loc[df.Survived == 0, ['Pclass', 'Age']]

Pclass   Age
0         3  22.0
4         3  35.0
5         3   NaN
6         1  54.0
7         3   2.0
..      ...   ...
884       3  25.0
885       3  39.0
886       2  27.0
888       3   NaN
890       3  32.0

[549 rows x 2 columns]

In [ ]:
df.loc[df.Survived == 0, ['Pclass', 'Age']] = 4
df.loc[df.Survived == 0, ['Pclass', 'Age']]

Pclass  Age
0         4  4.0
4         4  4.0
5         4  4.0
6         4  4.0
7         4  4.0
..      ...  ...
884       4  4.0
885       4  4.0
886       4  4.0
888       4  4.0
890       4  4.0

[549 rows x 2 columns]